File: android/app/src/main/kotlin/com/stemdeck/remote/library/LibraryFolder.kt
package com.stemdeck.remote.library

import java.util.UUID
import kotlinx.serialization.Serializable

/**
 * A user-created grouping in the song list (see [LibraryFolderStore]) —
 * purely a client-side organizational layer. StemDeck's own library has no
 * concept of folders, so this never round-trips to the server.
 */
@Serializable
data class LibraryFolder(
    val id: String = UUID.randomUUID().toString(),
    var name: String,
)

File: ios/StemDeckRemote/Library/LibraryFolder.swift
import Foundation

/// A user-created grouping in the song list (see `LibraryFolderStore`) —
/// purely a client-side organizational layer. StemDeck's own library has no
/// concept of folders, so this never round-trips to the server.
struct LibraryFolder: Codable, Identifiable, Equatable {
    let id: UUID
    var name: String

    init(id: UUID = UUID(), name: String) {
        self.id = id
        self.name = name
    }
}
